.github/workflows/sdk-node-format.yml

```diff
@@ -0,0 +1,42 @@
+name: Node SDK
+
+on:
+  pull_request:
+
+jobs:
+  format:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            changes:
+              - 'sdk/nodejs/'
+              - '.github/workflows/sdk-node-format.yml'
+
+      - if: steps.filter.outputs.changes == 'true'
+        uses: actions/checkout@v4
+
+      - if: steps.filter.outputs.changes == 'true'
+        name: Install Mise
+        run: |
+          curl https://mise.run | sh
+          echo "$HOME/.local/share/mise/bin" >> "$GITHUB_PATH"
+          echo "$HOME/.local/share/mise/shims" >> "$GITHUB_PATH"
+          mise settings experimental=true
+          mise install
+
+      - if: steps.filter.outputs.changes == 'true'
+        name: Caching
+        uses: actions/cache@v4
+        with:
+          key: node-sdk-${{ runner.os }}-${{ hashfiles('**/.mise.toml', './sdk/nodejs/**') }}
+          path: |
+            ./sdk/nodejs/node_modules
+            ~/.local/share/mise
+
+      - if: steps.filter.outputs.changes == 'true'
+        name: Run format
+        run: make format_check
```

Comment:
Now that we're introducing this, can we get rid of a different check?  Maybe not because this only runs on PRs, and not on merge etc.?

Reply:
> This PR doesn't contain any deletions from other jobs because we currently bundle linting and formatting into the same job for NodeJS, so this felt like a safe one to start with.

The plan was for this one to be the plan followed up by linting being the next one and the first one that requires actual coordination with required jobs etc